src/lib/update/clients/inst_update_partition_auto.rb

```diff
@@ -0,0 +1,154 @@
+# encoding: utf-8
+
+# ------------------------------------------------------------------------------
+# Copyright (c) 2016 SUSE LLC, All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# ------------------------------------------------------------------------------
+#
+# Authors:	Stefan Schubert <schubi@suse.de>
+#		Arvin Schnell <arvin@suse.de>
+#
+# Purpose:	Select root partition for update or booting.
+#		RootPart::rootPartitions must be filled before
+#		calling this module.
+#
+# $Id:$
+
+require "yaml"
+
+module Yast
+  class InstUpdatePartitionAutoClient < Client
```

Comment:
In this case is OK (it's an old client), but just as a reminder for the future see https://github.com/yast/yast-installation/pull/347#discussion_r55834329